components/style/traversal.rs

```diff
@@ -493,8 +497,15 @@ fn compute_style<E, D>(_traversal: &D,
             let shareable_element = {
                 // Perform the CSS selector matching.
                 context.thread_local.statistics.elements_matched += 1;
-                let filter = context.thread_local.bloom_filter.filter();
-                match_results = element.match_element(context, Some(filter));
+
+                // TODO(emilio): Here we'll need to support animation-only hints
+                // and similar.
+                match_results = if data.needs_only_style_attribute_update() {
```

Comment:
Given that it seems like we'll eventually want to do this the other way around, I think we want the opposite predicate here:

if data.restyle_self() {
   // do full selector matching
} else {
  if data.style_attribute_changed() {
  // Handle style attributes 
  }
  if data.tick_animation() {
  // Handle animations
  }
  // etc
}

So let's get rid of needs_only_style_attribute_update() (since it doesn't scale), and replace it with restyle_self() and style_attribute_changed().